src/app/app.component.ts

```diff
@@ -1,10 +1,33 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { filter, takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrl: './app.component.scss'
 })
-export class AppComponent {
+export class AppComponent implements OnInit, OnDestroy {
   title = 'TMS-Shop';
+  headerColor: 'black' | 'white' = 'white';
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(private route: ActivatedRoute, private router: Router) {}
+
+  ngOnInit(): void {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      takeUntil(this.ngUnsubscribe)
+    ).subscribe(() => {
+      const currentRouteData = this.route.snapshot.firstChild?.data;
+      this.headerColor = currentRouteData?.['headerColor'] || 'white';
```

Comment:
```suggestion
      this.headerColor = currentRouteData?.headerColor ?? 'white';
```

Reply:
Should be taken ['headerColor'] || better than ??